src/pybind/rados/rados.pyx

```diff
@@ -1109,6 +1111,45 @@ Rados object in state %s." % self.state)
         if ret < 0:
             raise make_ex(ret, "error creating pool '%s'" % pool_name)
 
+    @requires(('pool_name', str_type), ('crush_rule', opt(int)), ('auid', opt(int)))
+    def create_pool2(self, pool_name, crush_rule=None, auid=None):
```

Comment:
since `self.create_pool2(pool, rule)` is equivalent to `self.create_pool(pool, rule)`, why not extend the existing `create_pool(self, pool_name, crush_rule=None)` to `create_pool(self, pool_name, crush_rule=None, auid=None)`? as the latter will be backward compatible with the former.

Reply:
elif crush_rule is None and auid is not None:



            _auid = auid



            with nogil:



                ret = rados_pool_create_with_auid(self.cluster, _pool_name, _auid)

create_pool2 () has two more features than create_pool()，as follows.

1. create a pool owned by a specific auid
```
        elif crush_rule is None and auid is not None:
            _auid = auid
            with nogil:
                ret = rados_pool_create_with_auid(self.cluster, _pool_name, _auid)
```
2. create a pool with a specific CRUSH rule and auid
 ```
       else:
            _crush_rule = crush_rule
            _auid = auid
            with nogil:
                ret = rados_pool_create_with_all(self.cluster, _pool_name, _auid, _crush_rule)
```

Reply:
create_pool2 () has two more features than create_pool()，as follows.
1. create a pool owned by a specific auid
```
        elif crush_rule is None and auid is not None:
            _auid = auid
            with nogil:
                ret = rados_pool_create_with_auid(self.cluster, _pool_name, _auid)
```
2. create a pool with a specific CRUSH rule and auid
```
        else:
            _crush_rule = crush_rule
            _auid = auid
            with nogil:
                ret = rados_pool_create_with_all(self.cluster, _pool_name, _auid, _crush_rule)
```

Reply:
@zjcmszh i agree that create_pool2() is more feature-rich than create_pool(). and its features set covers that of create_pool(). that's why i suggest extend create_pool(), instead adding a new method.

Reply:
@tchaikov i could't agree with you more, i'll revise it when I get home at night.

Reply:
@tchaikov updated, please check.